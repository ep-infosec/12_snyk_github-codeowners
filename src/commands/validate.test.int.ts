import { v4 as uuidv4 } from 'uuid';
import fixtures, { invalidOwnerFixtures } from './__fixtures__/validate';
import { generateProject } from './__fixtures__/project-builder.test.helper';

import util from 'util';

const exec = util.promisify(require('child_process').exec);

describe('validate', () => {
  describe('when all owners are valid', () => {
    const testId = uuidv4();

    let testDir = 'not set';

    beforeAll(async () => {
      testDir = await generateProject(testId, fixtures);
      // tslint:disable-next-line:no-console
      console.log(`test scratch dir: ${testDir}`);
    });

    const runCli = async (args: string) => {
      return exec(`node  ../../../dist/cli.js ${args}`, { cwd: testDir });
    };

    it('output the result of all validation checks', async () => {
      const { stdout, stderr } = await runCli('validate');
      expect(stdout).toMatchSnapshot('stdout');
      expect(stderr).toMatchSnapshot('stderr');
    });
  });


  describe('when owners are invalid', () => {
    const testId = uuidv4();

    let testDir = 'not set';

    beforeAll(async () => {
      testDir = await generateProject(testId, invalidOwnerFixtures);
      // tslint:disable-next-line:no-console
      console.log(`test scratch dir: ${testDir}`);
    });

    const runCli = async (args: string) => {
      return exec(`node  ../../../dist/cli.js ${args}`, { cwd: testDir });
    };

    it('should throw on invalid users', async () => {
      await expect(() => runCli('validate')).rejects.toThrow();
    });
  });
});
