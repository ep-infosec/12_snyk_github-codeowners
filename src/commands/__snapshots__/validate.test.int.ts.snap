// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate when all owners are valid output the result of all validation checks: stderr 1`] = `
"Found duplicate rules [ 'duplicate-rule.js @some-owner' ]
Found rules which did not match any files [ 'file-does-not-exist.md @some-owner' ]
"
`;

exports[`validate when all owners are valid output the result of all validation checks: stdout 1`] = `""`;
